### PR TITLE
app service uses subnet

### PR DESCRIPTION
**Pull request by WillBeebe**
These changes are from me testing a few apps on Azure. 

- enables logging for app service apps
- creates a subnet so that the app service can connect to things in the VNet

resolves: https://github.com/massdriver-cloud/application-templates/issues/51

### CODE REVIEW

File: massdriver-application-azure-app-service/_variables.tf [hunk 1]
```diff
@@ -29,3 +29,17 @@ variable "tags" {
 variable "contact_email" {
   type = string
 }
+
+variable "command" {
+  type    = string
+  default = null
+}
+
+variable "virtual_network_id" {
```
Comment (WillBeebe) [inline massdriver-application-azure-app-service/_variables.tf:38]: needed to create the subnet

File: massdriver-application-azure-app-service/_variables.tf [hunk 2]
```diff
@@ -29,3 +29,17 @@ variable "tags" {
 variable "contact_email" {
   type = string
 }
+
+variable "command" {
```
Comment (WillBeebe) [inline massdriver-application-azure-app-service/_variables.tf:33]: adding container command. if blank, we use the default like k8s

File: massdriver-application-azure-app-service/auto-scaling.tf [hunk 3]
```diff
@@ -0,0 +1,66 @@
+resource "azurerm_monitor_autoscale_setting" "main" {
```
Comment (WillBeebe) [inline massdriver-application-azure-app-service/auto-scaling.tf:1]: moved so grokking `main.tf` is easier

File: massdriver-application-azure-app-service/subnet.tf [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  delegation {
+    name = "virtual-network-integration"
+
+    service_delegation {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
```
Comment (mclacore) [inline massdriver-application-azure-app-service/subnet.tf:44]: Opened a [PR](https://github.com/massdriver-cloud/terraform-modules/pull/60) to add default network security group to this subnet.

**Review by mclacore — CHANGES_REQUESTED** [review]

File: massdriver-application-azure-app-service/subnet.tf [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+# Isolated tier (max?): 100
+# So we need 105
+# But when scaling, max is doubled.... so 210
+# CIDR minus 5, halved and floored
+# /28 16 - 5 = 11 => 5
+# /27 32 - 5 = 27 => 13
+# /26 64 - 5 = 59 => 29
+# /25 128 - 5 = 123 => 61
+# /24 256 - 5 = 251 => 125
+# Ideally, Massdriver picks an IP range that is not in use
```
Comment (WillBeebe) [inline massdriver-application-azure-app-service/subnet.tf:28]: Update to explain why we required a `/24`

File: massdriver-application-azure-app-service/subnet.tf (hunk 4 above)
Comment (mclacore) [reply massdriver-application-azure-app-service/subnet.tf:44]: Disregard ^

**Review by xpositivityx — APPROVED** [review]

**Pull request merged**